lib/ansible/modules/system/macos_pref.py

```diff
@@ -0,0 +1,372 @@
+- name: Set desktop and finder icon view settings
+  macos_pref:
+    domain: com.apple.finder
+    key: FK_StandardViewSettings
+    value:
+      IconViewSettings:
+        arrangeBy: name
+        gridSpacing: 44
+        iconSize: 36
+        showIconPreview: true
+        showItemInfo: false
+        labelOnBottom: true
+        textSize: 12
+
+
+# Set (merge) a dictionary key
+- name: Set desktop and finder icon size
+  macos_pref:
+    domain: com.apple.finder
+    key: FK_StandardViewSettings
+    value:
+      IconViewSettings:
+        iconSize: 50
+    state: merge
+
+
+# Read a dictionary key
+- name: Show all files
+  macos_pref:
+    domain: com.apple.finder
+    key: FK_StandardViewSettings
+  register: finder_view_settings
+
+- name: Check show all files
+  fail: msg="Only iconSize=50 is acceptable!"
+  when: finder_view_settings.value.IconViewSettings.iconSize != 50
+
+
+# Delete a key
+- name: Set Terminal default window settings
+  macos_pref:
+    domain: com.apple.Terminal
+    key: Default Window Settings
+    state: absent 
+
+
+# Read directly from a plist file (also works for writing)
+- name: Read rumour from file
+  macos_pref:
+    domain: /tmp/rumours.plist
+    key: Rumour
+  register: rumour
+
+
+# Set a dict on currentHost
+- name: Set screensaver
+  macos_pref:
+    domain: com.apple.screensaver
+    host: currentHost
+    key: moduleDict
+    value:
+      moduleName: Flurry
+      displayName: Flurry
+      path: /System/Library/Screen Savers/Flurry.saver
+      type: 0
+'''
+
+RETURN = '''
+value:
+    description: The value associated with the preference domain and key.
+                 Return type is a python object that maps closest to the data type of the macOS preference.
+                 This can be an integer, float, string, dict, list, etc...
+    type: string
+    returned: on success
+    sample: "{'CustomViewStyleVersion': 1}"
+'''
+
+import collections
+import copy
+import sys
+from ansible.module_utils.basic import AnsibleModule
+
+# Ensure that we have access to the pyobjc libraries that Apple ships with
+# macOS (e.g. even when using brew-installed python)
+try:
+    sys.path.insert(0, '/System/Library/Frameworks/Python.framework/Versions/2.7/Extras/lib/python/PyObjc')
+    import CoreFoundation
+    from Foundation import kCFPreferencesCurrentUser, kCFPreferencesAnyUser, kCFPreferencesCurrentHost, kCFPreferencesAnyHost
+    from PyObjCTools.Conversion import pythonCollectionFromPropertyList
+except ImportError:
+    pyobjc_found = False
+else:
+    pyobjc_found = True
+
+
+class MacOSPrefException(Exception):
+    pass
+
+
+class MacOSPref(object):
+
+    global pyobjc_found
+
+    def __init__(self, **kwargs):
+        # Exit if PyObjC module is not available
+        if not pyobjc_found:
+            raise MacOSPrefException("The PyObjC python module is required.")
+
+        # Set all given parameters
+        for key, val in kwargs.items():
+            setattr(self, key, val)
```

Comment:
This looks a bit dangerous to me, and does mean a person has to figure out what parameters need to be passed-in into the constructor based on existing calling code. I would recommend listing the possible argument to `__init__` explicitly. Processing `kwargs` in this way should probably be left for cases where the implementation actually deals with variable set of parameters.

Reply:
Will update that 👍